GitCommands/Git/AheadBehindDataProvider.cs

```diff
@@ -44,17 +46,20 @@ public void ResetCache()
                 return null;
             }
 
-            // Callers setting branch name has the responsibility to ensure that not all are needed
-            if (string.IsNullOrWhiteSpace(branchName) && !string.IsNullOrWhiteSpace(_branchName))
+            lock (_lock)
```

Comment:
_lazyData could be init earler (but more complex), with the lock lazy could be removed.
Not worth changing...